app/org/sagebionetworks/bridge/cache/CacheProvider.java

```diff
@@ -26,41 +34,72 @@ public void setJedisOps(JedisOps jedisOps) {
         this.jedisOps = jedisOps;
     }
 
-    public void setUserSession(String key, UserSession session) {
+    public void setUserSession(final UserSession session) {
+        checkNotNull(session);
+        List<Object> results = null;
         try {
-            String ser = BridgeObjectMapper.get().writeValueAsString(session);
-            String redisKey = RedisKey.SESSION.getRedisKey(key);
-            String result = jedisOps.setex(redisKey, BridgeConstants.BRIDGE_SESSION_EXPIRE_IN_SECONDS, ser);
+            final String userId = session.getUser().getId();
+            final String userKey = RedisKey.USER_SESSION.getRedisKey(userId);
+            final String sessionToken = session.getSessionToken();
+            final String sessionKey = RedisKey.SESSION.getRedisKey(sessionToken);
+            final String ser = BridgeObjectMapper.get().writeValueAsString(session);
+            results = jedisOps.getTransaction()
+                    .setex(userKey, BridgeConstants.BRIDGE_SESSION_EXPIRE_IN_SECONDS, sessionToken)
```

Comment:
Users can create more than one session, but when they sign out they'll be under one GUID only, are you keeping a list of all the GUID session keys for the given user?

To have one session per user no matter how many times they sign in, we could store the session under the user ID, and then one more entry mapping the GUID to the user ID. On sign in you could then find out immediately if that user already had a session, and return the guid for that. Or find the session by the GUID with a double look-up.